### PR TITLE
AboutHashtables Fixes

### LINKED ISSUE #169: Issue with AboutHashtables.koans.ps1 file
# Describe "Koan Bug, Issue, or Help Request"

AboutHashtables.koans.ps1
This might not be a serious problem, but solving it helps having an unambiguous code.
 
## Context "The Problematic Assertions"

        It 'will not implicitly convert keys and lookup values' {
            $Hashtable = @{0 = 'Zero'}

            $Hashtable[0] | Should -Be 'Zero'
            $Hashtable['0'] | Should -Be '__'
        }

## Context "Your Attempts"

        It 'will not implicitly convert keys and lookup values' {
            $Hashtable = @{0 = 'Zero'}

            $Hashtable[0] | Should -Be 'Zero'
            $Hashtable['0'] | Should -Be __
        }

## Context "Additional Information"

The single quotes around the __ must be removed, because we need to provide the $null in its place and with quotation marks around it the $null would be considered as a string. Sorry if I didn't put the codes in the right places, after all this is my first issue report on GitHub ever :)

### LINKED ISSUE #170: Hashtable keys are not always in the same order
# Describe "Koan Bug, Issue, or Help Request"

https://github.com/vexx32/PSKoans/blob/e0a54b0363c84ef67262a34dad48caa9fd745e02/PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1#L109

This seems to have been solved once already in #124, but it is back again. Hashtable key hashes seem to be tied to the session, so order is different every session. 

## Context "The Problematic Assertions"

It 'allows you to retrieve a list of keys or values'
It 'is not ordered'
It 'allows you to remove keys'

## Context "Your Attempts"

Ref this https://clips.twitch.tv/CharmingSmoothRedpandaImGlitch

### PR DESCRIPTION
**Pull request by vexx32**
﻿# PR Summary

Some refactoring and updates to `AboutHashtables.Koans.ps1`

Resolves #169 
Resolves #170

/cc @arefg @tomlarse -- if either of you fellows have time to give this a look over and check if I missed anything, I'd greatly appreciate it! 😊 

## Context

See linked issues.

## Changes

- Avoid hashtable order of keys/values problems
- Add some better comments
- Refactor to have fill-in values on LHS to provide more useful error messages in the meditation prompt.

## Checklist

- [X] Pull Request has a meaningful title.
- [X] Summarised changes.
- [X] Pull Request is ready to merge & is not WIP.
- [ ] ~~Added tests / only testable interactively.~~
  - ~~Make sure you add a new test if old tests do not effectively test the code changed.~~
- [ ] ~~Added documentation / opened issue to track adding documentation at a later date.~~

### CODE REVIEW

File: PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1 [hunk 1]
```diff
@@ -104,15 +103,15 @@ Describe 'Hashtables' {
         }
 
         It 'allows you to retrieve a list of keys or values' {
-            $Hashtable = @{One = 1; Two = 2; Three = 3; Four = 4}
+            $Hashtable = @{One = 1; Two = 2; Three = 3; Four = 4 }
 
             $Hashtable.Keys | Should -Be @('__', '__', '__', '__')
             $Hashtable.Values | Should -Be @( )
```
Comment (tomlarse) [inline PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1]: The order of the keys here will still be different on every new session, so if you close your session and try to continue later, you will have to solve these again to move on.

File: PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1 [hunk 2]
```diff
@@ -96,45 +101,58 @@ Describe 'Hashtables' {
...
-            $Hashtable.Values | Should -Be @(1, 2, 3, 4, 5)
+                The order can and will change again, as well, if the collection is altered:
+
+                    $Hashtable['Five'] = 5
+            #>
+
+            $Hashtable.Keys | Should -Not -Be @('One', 'Two', 'Three', 'Four', 'Five')
+            $Hashtable.Values | Should -Not -Be @(1, 2, 3, 4, 5)
+
+            $Hashtable.Keys | Should -BeIn @('____', 'Two', '____', 'Four', 'Five')
```
Comment (tomlarse) [inline PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1]: This looks good!

File: PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1 [hunk 3]
```diff
@@ -97,45 +101,58 @@ Describe 'Hashtables' {
...
-            $Hashtable['Five'] = 5
+                    $Hashtable['Five'] = 5
+            #>
 
-            $Hashtable.Keys | Should -Be @('One', 'Two', 'Three', 'Four', 'Five')
-            $Hashtable.Values | Should -Be @(1, 2, 3, 4, 5)
+            $Hashtable.Keys | Should -Not -Be @('One', 'Two', 'Three', 'Four', 'Five')
+            $Hashtable.Values | Should -Not -Be @(1, 2, 3, 4, 5)
+
+            $Hashtable.Keys | Should -BeIn @('____', 'Two', '____', 'Four', 'Five')
```
[the remark this answers is not in the document]
Comment (tomlarse) [reply PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1]: Good catch, the `$Hashtable['Five'] = 5` should probably not be in the comments.

File: PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1 (hunk 2 above)
Comment (vexx32) [reply PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1]: I think the chances are sufficiently low, but if we see it become an issue we can relegate those two to comments instead I suppose. 🙂

File: PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (vexx32) [reply PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1]: Good call, I must have missed that when I refactored earlier.

**Pull request merged**